presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -239,6 +246,43 @@ private boolean nextRow(MessageAndOffset messageAndOffset)
             return true; // Advanced successfully.
         }
 
+        private void openFetchRequest()
+        {
+            if (messageAndOffsetIterator == null) {
+                log.info("Fetching %d bytes from partition %d @offset %d (%d - %d) -- %d messages read so far",
+                        fetchSize, split.getPartitionId(), cursorOffset, split.getStart(), split.getEnd(), totalMessages);
+                cursorOffset += fetchedSize;
+                FetchRequest req = new FetchRequest(split.getTopicName(), split.getPartitionId(), cursorOffset, fetchSize);
```

Comment:
Xiaoyao mentioned c7717f637949180bd421487d36ba6db8565292e8 is copying master connector and f621dcca1f62643fd8f0bb31770efad14b848657 is our edits.